docs/07_code_coverage_artifact_import.md

```diff
@@ -0,0 +1,347 @@
+  ```text
+  {
+    // ...
+    "build_num": <integer>,
+    "has_artifacts": <boolean>,
+    // ...
+  }
+  ```
+- **GET** `api/v1.1/project/{project_slug}/{build_number}/artifacts` responses with a list of artifacts generated during a build specified by a `build_number`. Each artifact has the `url` field that contains a download URL for the artifact.
+  ```text
+  {
+    // ...
+    "path": "coverage/coverage.json",
+    "url": "https://.../coverage/coverage.json"
+  }
+  ```
+
+More information about the `project_slug` parameter can be found [here](https://circleci.com/docs/2.0/api-developers-guide/#getting-started-with-the-api). CircleCI API v1.1 (current stable version) [Summary](https://circleci.com/docs/api/#summary-of-api-endpoints) and [Overview](https://circleci.com/docs/api/#api-overview).
+
+# Jenkins
+
+## Deploy artifacts
+Jenkins build job can be configured to store generated artifacts as follows:
+1. In `Jenkinsfile` go to a stage you want to configure artifacts uploading for.
+2. Add the `post` section to the stage.
+3. Add `archiveArtifacts` step to the created `post` section with path to file or path pattern (see example below).
+
+```text
+pipeline {
+    agent any
+    stages {
+        stage('Build') {
+            steps {
+                sh 'bash coverage.sh'
+            }
+            post {
+                always {
+                    archiveArtifacts 'coverage/*'
+                }
+            }
+        }
+    }
+}
+```
+
+For more information in storing artifacts follow Jenkins [documentation](https://jenkins.io/doc/) and this [tour](https://jenkins.io/doc/pipeline/tour/tests-and-artifacts/)
+
+## Create API Access Token
+To access Jenkins's endpoints the authorization is required. You can use your username and password you've registered within an HTTP Basic Auth. For example:
+1. Base64 encode username and password string as `username:password`.
+2. Pass the result to `Authorization` header as `Authorization: Basic <token>`.
+
+## Retrieve artifacts via API
+Here are Jenkins' endpoints to work with builds and artifacts:
+- **GET** `{jenkins_url}/api/json` responses with a master node of Jenkins that contains common information and list of jobs (aka Pipelines) available.
+  ```text
+  {
+    // ...
+    "jobs": [
+      {
+        "name": <string>,
+        // ...
+      },
+      // ...
+    ]
+  }
+  ```
+  Alternatively, you can request **GET** `{jenkins_url}/api/json?tree=jobs[name]` to fetch only list of jobs available.
+  ```text
+  {
+    "jobs": [
+      {
+        "name": <string>
+      },
+      // ...
+    ]
+  }
+  ```
+  Select desired job and proceed to next endpoints using its `name` field.
+- **GET** `{jenkins_url}/job/{job_name}/api/json` responses with a job common information that contains a list of jobs (aka Workflows) available.
```

Comment:
We seem to use jobs, workflows, pipelines here interchangeably should we clear it somehow?